### ISSUE TITLE
vault 0.8 leaves plugins dangling on Windows

### ISSUE THREAD

**Issue by issacg** [hashicorp/vault#3136]
**Environment:**
Vault v0.8.0 ('af63d879130d2ee292f09257571d371100a513eb')

* Vault Version:
0.8 Win64

**Startup Log Output:**
2017/08/09 20:46:23.394742 [TRACE] database: operation=Close status=started type=plugin-redshift
2017-08-09T20:46:23.414+0300 [WARN ] plugin: error closing client during Kill: err="unexpected EOF"
2017/08/09 20:46:23.422063 [TRACE] database: operation=Close status=finished type=plugin-redshift err=<nil> took=27.3216ms
2017/08/09 20:46:23.425988 [INFO ] core: pre-seal teardown complete
2017/08/09 20:46:23.427951 [DEBUG] core: sealing barrier
2017/08/09 20:46:23.428932 [INFO ] core: vault is sealed

**Expected Behavior:**
On shutdown plugins should terminate

**Actual Behavior:**
Plugins no longer terminate (this worked as expected in 0.7.3)

**Steps to Reproduce:**
Use a plugin with Vault.  The above is using a database plugin

**Important Factoids:**
This worked fine on this plugin in 0.7.3  I just rebuilt and tweaked to use the new plugin API, and now I need to manually kill the plugin process even if I kill vault.  This was only reproduced in my development environment so far (I haven't had anywhere else to easily test)

**Comment by calvn** [comment]
How are you running vault? I was able to reproduce the log message and got the [WARN ] log from the plugin process, but did not end up with a dangling process on Windows using PowerShell. The only case where I observed a dangling process was if I was running vault in cygwin and issued a SIGINT/SIGTERM, but in that scenario vault didn't seem to shutdown properly (i.e. there was no "Vault shutdown triggered" message or any of the cleanup logs thereafter).
